### PR TITLE
Fixed the like button count issue

### LINKED ISSUE #1725: Like functionality is not working for searched output for People tab
Staging env - Tested on Android 9 Mobile and Windows 10 laptop on Chrome Browser

https://www.hippovideo.io/video/view/6758057

### PR DESCRIPTION
**Pull request by mathigayathri2001**
### What does this pull request aim to achieve? 💯
This PR fixed the like button issue
_Please be concise and link any related issue(s):_This PR resolves #1725 

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x ] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Review by rabbi08 — CHANGES_REQUESTED** [review]
The user is able to like his own post but cannot like other user's post, see below GIF.

ezgif com-gif-maker (42)

**Comment by mathigayathri2001** [comment]
> The user is able to like his own post but cannot like other user's post, see below GIF.
> 
> ezgif com-gif-maker (42)

Thanks for the feedback, I will look into it.

**Comment by mathigayathri2001** [comment]
@rabbi08  I fixed the issue . Now, user can like the other users post   . Please check it ,please note that i formatted the entire code , it looks like some  lines of the code format changed automatically which i did not change . if it is a issue i can roll back that format . Please let me know .

**Comment by rabbi08** [comment]
@mathigayathri2001 
Verified the issue on feature build and it is working fine. Thanks.
please see below GIF:

ezgif com-gif-maker (47)

**Review by rabbi08 — APPROVED** [review]

File: client/src/pages/Profile.js [hunk 1]
```diff
@@ -349,12 +349,19 @@ const Profile = ({
 
   const handlePostLike = async (postId, liked, create) => {
     sessionStorage.removeItem("likePost");
+    let currentUser;
+    try {
+      const res = await axios.get("/api/users/current");
+      currentUser = res.data.id;
+    } catch (error) {
+      console.log({ error });
+    }
```
Comment (Zaydme) [inline client/src/pages/Profile.js:358]: I don't think we will need an API call to get the user id here.
Why is the `user._id` not working in the first place in this specific case, I think there might be a much simpler solution that doesn't require an extra API call.

**Review by Zaydme — CHANGES_REQUESTED** [review]

File: client/src/pages/Profile.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ const URLS = {
   website: [websiteIcon],
 };
 
-const getHref = (url) => (url.startsWith("http") ? url : `//${url}`);
+const getHref = url => (url.startsWith("http") ? url : `//${url}`);
```
Comment (joshmorel) [inline client/src/pages/Profile.js:106]: Why all these format changes? It seems like you are using a different linter or prettier config.

File: client/src/pages/Profile.js (hunk 1 above)
Comment (joshmorel) [reply client/src/pages/Profile.js:358]: Good catch @Zaydme .

The issue was they were trying to use the same user as the profile so of course it was forbidden:

image

However, we are completely changing `handlePostLike` with #1767 and it is working there as expected. Thanks for the contribution @mathigayathri2001, but I don't think it's worth it to fix these issues and re-review. I would propose we close this & aim to get #1767 working, reviewed & merged.

**Review by joshmorel — CHANGES_REQUESTED** [review]

File: client/src/pages/Profile.js (hunk 2 above)
Comment (mathigayathri2001) [reply client/src/pages/Profile.js:106]: Yes, during development I have used prettier standard javascript format, I did not noticed before pushing the code. So, I mentioned @rabbi08 in the earlier message. Sorry about that, hereafter I will make sure about it.

File: client/src/pages/Profile.js (hunk 1 above)
Comment (mathigayathri2001) [reply client/src/pages/Profile.js:358]: Thanks for the feedback.

**Comment by rabbi08** [comment]
This issue is fixed with PR#1767, hence closing this one.
Thanks @mathigayathri2001

**Pull request closed without merging**